src/postgres/getClientConfig.ts

```diff
@@ -17,21 +17,36 @@ export async function getClientConfig(treeItem: PostgresServerTreeItem, database
 
     if (!(username && password)) {
         const credentials = await treeItem.getCredentials();
-        username = nonNullProp(credentials, 'username');
-        password = nonNullProp(credentials, 'password');
+        if (credentials.username && credentials.password) {
+            username = credentials.username;
+            password = credentials.password;
+        }
     }
 
-    const ssl: ConnectionOptions = {
+    const sslAzure: ConnectionOptions = {
         // Always provide the certificate since it is accepted even when SSL is disabled
         // Certificate source: https://aka.ms/AA7wnvl
         ca: BaltimoreCyberTrustRoot
     };
 
-    if (username && password) {
+    if ((username && password) || username === 'postgres') {
         const host = nonNullProp(treeItem.connectionString, 'hostName');
         const port: number = treeItem.connectionString.port ? parseInt(treeItem.connectionString.port) : postgresDefaultPort;
-        const clientConfig = { user: username, password: password, ssl, host, port, database: databaseName };
-        const client = new Client(clientConfig);
+        let ssl: boolean | ConnectionOptions | undefined;
+        if (treeItem.azureName) {
+            ssl = sslAzure;
+        } else {
+            ssl = treeItem.connectionString.ssl;
+        }
+        const clientConfig: ClientConfig = { user: username, password: password, ssl: ssl, host, port, database: databaseName };
+        let client: Client;
+        if (treeItem.azureName) {
+            client = new Client(clientConfig);
+        } else {
+            client = new Client({
+                connectionString: treeItem.connectionString.connectionString,
```

Comment:
So this line is good - I like directly using the connectionString for the attached scenario. However, the problem is that the client created here is never returned or used except for a few lines down. That could cause problems if the `ClientConfig` returned by this function is different than what's specified in the connection string. I think the best solution is to refactor `getClientConfig` to `createPgClient`. Something like this:

```
export async function createPgClient(treeItem: PostgresServerTreeItem, databaseName: string): Promise<Client> {
    if (treeItem.azureName) {
        // all the stuff to create clientConfig for the Azure case
        const clientConfig: ClientConfig = { user: username, password: password, ssl: ssl, host, port, database: databaseName };
        return new Client(clientConfig);
    } else {
        return new Client(treeItem.connectionString.connectionString);
    }
}
```

That way, it's the "pg" module's job to create a ClientConfig from the connection string, not us.

Reply:
Passing Client object will cause problems as Client can only be connected once. Also, for functions and tables the same client will be passed down. We can use Pool which will allow for multiple client connections but it will require a lot more code change and each client needs to be released to make sure there is no lag and client availability is guaranteed. 

We can still return ClientConfig for both cases, with the exception that ClientConfig for attach accounts scenario would only contain `connectionString:`.

Reply:
> We can still return ClientConfig for both cases, with the exception that ClientConfig for attach accounts scenario would only contain connectionString:.

Yeah that'll work. I still think the above proposed solution is better (and I'll explain why below), but I agree this is sufficient for this PR and would require fewer changes.

> Passing Client object will cause problems as Client can only be connected once.

It's a bit excessive to call `client.connect` every time we call `getClientConfig`. We should only need to do it once per database, probably as a part of `checkAuthentication`.

> Also, for functions and tables the same client will be passed down.

I don't think `clientConfig` should be passed down in the first place. It would be better if the child tree items (like functions/tables) called something like `this.parent.getClientConfig()` to make sure they always get the latest clientConfig. Today, the parent might've changed clientConfig with a different username/password and the children may not get that unless the user refreshes the tree.

My point is that the two obstacles you mentioned should probably be refactored eventually anyways.

Reply:
Yes I agree with you on that, the way we are connecting to client and getting clientConfig is not systematic and there could gaps in this process. I will create an issue for it for next release.